### ISSUE TITLE
bug: SKIP_SSL_CERT_DOWNLOAD=1 not being applied to Lambda Docker

### ISSUE THREAD

**Issue by cmitchell** [localstack/localstack#8932]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I am on an isolated network, with no access to the internet. We have a private docker mirror, so pulling images is possible, but there seems to be quite a bit of dependency on having a  connection to the internet. The following docker-compose.yml allows localstack to come up cleanly because it skips certificate verification, and for me to deploy a lambda function it cleanly, but when I invoke it, the lambda container is attempting to verify the certificates.

I have hand  typed the following, so please excuse typos.

```
version: '3.8'

services:
  localstack:
    image: localstack/localstack
    privileged: true
    ports:
      - "127.0.0.1:4566:4566"
      - "127.0.0.1:4510-4559:4510-4559"
    environment:
      - SERVICES=lambda
      - DEBUG=1
      - SKIP_SSL_CERT_DOWNLOAD=1
      - LAMBDA_DOCKER_FLAGS="-e SKIP_SSL_CERT_DOWNLOAD=1"   <-- trying to pass to the lambda localstack, doesn't work
      - DOCKER_FLAGS="-e SKIP_SSL_CERT_DOWNLOAD=1"    <-- another attempt to pass to the lambda localstack, doesn't work
      - DATA_HOST=unix:///var/run/docker.sock
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      - ${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstrack
```

Running `docker-compose up -d` doesn't complain. Inspecting the `docker logs <container-id>` shows the following 
LocalStack version: 2.2.1.dev
LocalStack build date: 2023-07-23
LocalStack build git hash: 6f846a82

I can also see in the logs 
"Skipping download of local SSL cert, as SKiP_SSL_CERT_DOWNLOAD=1"

I can also see it pull the lambda runtime image (that I had added to our mirror
"Pulling Docker image: public.ecr.aws/lambda/java17"

Upon running `serverless invoke --function ourLambdaFunction -d {"our-data": "blah blah blah"}`  The first sign of trouble in the logs is
"l.u.c.container_networking:   Unable to get main container IP address: localstack_main"
"plugin.manager:    no extensions  found in namespace localstack.hooks.lambda_start_docker_executor"
-- docker_sdk_client creates the container with a bunch of attributes that I'm not re-typing, but none of them  are about skipping ssl cert verficiation
"localstack.packages.api :    Starting installation of awslambda-runtime..."
"l.s.s.i.runtime_environment:  Failed to start runtime environment for ID=XXXXXXXXXXXX with: Installation of awslambda-runtime failed."
Then there is a python stack coming out of /opt/code/localstack/...." with the message at the end of the stack being
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILURE] certificate verify failed: unable to get local issuer certificate.
Inspecting the stack trace also shows that it's attemping MyHTTPSConnectionPool(host='github.com', port=443: Max retries executed with url: /localstack/lambda-runtime-init/releases/download/v0.1.19-pre/aws-lambda-rie-x68_64, which is the confirmation that it's still trying to reach the normal internet.

Is there another configuration option that I need to add or a way to pass through the option to the lambda docker container? 
I don't think it's getting to my lambda code because it's not getting passed starting the runtime, but I can post my serverless.yml if you think it would help to reproduce. Thanks for your help.

### Expected Behavior

That the lambda container (that I can't control) would also have ssl verification disabled .

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker-compose up

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    serverless deploy --stage dev   (works)
    serverless invoke --function  <ourFunction> -d <our data>

### Environment

```markdown
- OS:   amazon linux
- LocalStack: 2.2.1.dev
```

**Comment by dfangl** [comment]
Hello!
Lambda has a runtime dependency it will download automatically if not locally present. This dependency is needed for lambda execution.

However, since lambda is an important service for us, we recently, with 1239d4d , started including it in the image. Can you make sure your docker mirror has the latest image, so this change is included?

**Comment by cmitchell** [comment]
@dfangl   I see that the image we have is 25 days old, so I have requested that an updated latest tag be pulled in. Hopefully I can try it out later today.

Just a point of clarification though, I already ran into the issue of it trying to download the runtime, so I had the public.ecr.aws/lambda/java17  image pulled in. It found it successfully. Before I did that, I couldn't even deploy. Where the issue comes from is that the lambda runtime image tries to get certificates from github, and I couldn't find a way to disable that behavior. Does the runtime that is included in the latest localstack still try to reach out to github for certificates?
